src/Host/Client/Impl/Host/RemoteBrokerClient.cs

```diff
@@ -47,13 +47,13 @@ public RemoteBrokerClient(string name, IRSessionProvider sessionProvider, Broker
 
         public override async Task<RHost> ConnectAsync(HostConnectionInfo connectionInfo, CancellationToken cancellationToken = new CancellationToken()) {
             var host = await base.ConnectAsync(connectionInfo, cancellationToken);
-
-            var aboutHost = await GetHostInformationAsync<AboutHost>(cancellationToken);
-            var brokerIncompatibleMessage = aboutHost?.IsHostVersionCompatible();
-            if (brokerIncompatibleMessage != null) {
-                throw new RHostDisconnectedException(brokerIncompatibleMessage);
+            if (connectionInfo.IsInteractive) {
+                var aboutHost = await GetHostInformationAsync<AboutHost>(cancellationToken);
+                var brokerVersionMessage = aboutHost?.IsHostVersionCompatible();
+                if (brokerVersionMessage != null) {
+                    _console.WriteError(brokerVersionMessage);
```

Comment:
We don't handle protocol incompatibility anywhere. If we're still returning a host, this will be broken in a different place.